lib/calendar.ts

```diff
@@ -2362,24 +2376,26 @@ class NonIsoCalendar implements CalendarImpl {
     const cache = new OneObjectCache();
     // For lunisolar calendars, either `monthCode` or `year` must be provided
     // because `month` is ambiguous without a year or a code.
-    const fieldNames = this.fields(['day', 'month', 'monthCode', 'year']) as AnyTemporalKey[];
-    const fields = ES.PrepareTemporalFields(fieldsParam, fieldNames, []);
+    const fieldNames = ['day', 'month', 'monthCode', 'year'] as AnyTemporalKey[];
+    const extraFieldDescriptors = this.CalendarFieldDescriptors('date');
+    const fields = ES.PrepareTemporalFields(fieldsParam, fieldNames, [], extraFieldDescriptors);
     const overflow = ES.ToTemporalOverflow(options);
     const { year, month, day } = this.helper.monthDayFromFields(fields, overflow, cache);
     // `year` is a reference year where this month/day exists in this calendar
     const result = ES.CreateTemporalMonthDay(month, day, calendarSlotValue, /* referenceISOYear = */ year);
     cache.setObject(result);
     return result;
   }
-  fields(fieldsParam: string[]): string[] {
-    let fields = fieldsParam;
-    if (ArrayIncludes.call(fields, 'year')) fields = [...fields, 'era', 'eraYear'];
+  fields(fields: string[]): string[] {
+    if (ES.Call(ArrayIncludes, fields, ['year'])) {
+      ES.Call(ArrayPrototypePush, fields, ['era', 'eraYear']);
```

Comment:
I think that this is actually OK, because `Calendar.protototype.fields` creates a new array before passing it to this method. 

https://github.com/js-temporal/temporal-polyfill/blob/990d51a793f898a39457cb5e14b11eadeda1fb2e/lib/calendar.ts#L187

Reply:
I added a comment to clarify that mutation is safe, and will resolve this thread. 

```ts
  fields(fields: string[]): string[] {
    // Note that `fields` is a new array created by the caller of this method,
    // not the original input passed by the original caller. So it's safe to
    // mutate it here because the mutation is not observable.
    if (ES.Call(ArrayIncludes, fields, ['year'])) {
      ES.Call(ArrayPrototypePush, fields, ['era', 'eraYear']);
    }
    return fields;
  }
 ```